eth-contracts/contracts/DelegateManager.sol

```diff
@@ -445,16 +441,11 @@ contract DelegateManager is InitializableV2 {
         );
 
         // Rewards directly allocated to service provider for their stake
-        // Total active funds for direct deployer reward share
-        /// totalActiveDeployerFunds = totalBalanceInSPFactory.sub(spLockedStake);
-        uint256 spRewardShare = (
-            (totalBalanceInSPFactory.sub(spLockedStake)).mul(totalRewards)
-        ).div(totalActiveFunds);
-
+        uint256 spRewardShare = totalRewards.sub(totalDelegatedStakeIncrease);
         spFactory.updateServiceProviderStake(
             _serviceProvider,
-            /// newSpBalance = totalBalanceInSPFactory + spRewardShare + spDeployerCutRewards;
-            totalBalanceInSPFactory.add(spRewardShare.add(spDeployerCutRewards))
+            /// newSpBalance = totalBalanceInSPFactory + spRewardShare
+            totalBalanceInSPFactory.add(spRewardShare)
```

Comment:
@roneilr wdyt about adding a final assert on the sum of totalInSpfactory + totalInDelMan = totalInstaking?

Reply:
i think that's a great idea - that invariant must always be satisfied